app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/Stepper.kt

```diff
@@ -0,0 +1,693 @@
+    } else if (!isMinusEnabled) {
+      minusColor.animateTo(pulsatingSymbolColor, animationSpec = enterAnimationSpec)
+      minusColor.animateTo(disabledSymbolColor, animationSpec = exitAnimationSpec)
+    } else if (showError) {
+      minusColor.animateTo(pulsatingSymbolColor, animationSpec = enterAnimationSpec)
+      minusColor.animateTo(enabledSymbolColor, animationSpec = exitAnimationSpec)
+    } else {
+      minusColor.animateTo(enabledSymbolColor, animationSpec = exitAnimationSpec)
+    }
+  }
+  LaunchedEffect(showError) {
+    if (showError) {
+      launch {
+        containerColor.animateTo(pulsatingContainerColor, animationSpec = enterAnimationSpec)
+        containerColor.animateTo(initialContainerColor, animationSpec = exitAnimationSpec)
+      }
+      launch {
+        labelColor.animateTo(pulsatingLabelColor, animationSpec = enterAnimationSpec)
+        labelColor.animateTo(initialLabelColor, animationSpec = exitAnimationSpec)
+      }
+      launch {
+        textColor.animateTo(pulsatingTextColor, animationSpec = enterAnimationSpec)
+        textColor.animateTo(initialTextColor, animationSpec = exitAnimationSpec)
+      }
+    } else {
+      launch { containerColor.animateTo(initialContainerColor, animationSpec = enterAnimationSpec) }
+      launch { labelColor.animateTo(initialLabelColor, animationSpec = enterAnimationSpec) }
+      launch { textColor.animateTo(initialTextColor, animationSpec = enterAnimationSpec) }
+    }
+  }
+  when (stepperStyle) {
+    Default -> DefaultStepper(
+      stepperSize = stepperSize,
+      text = text,
+      onMinusClick = onMinusClick,
+      onPlusClick = onPlusClick,
+      containerColor = containerColor.value,
+      textColor = textColor.value,
+      plusColor = plusColor.value,
+      minusColor = minusColor.value,
+      errorText = errorText,
+      showError = showError,
+      modifier = modifier,
+    )
+
+    is Labeled -> LabeledStepper(
+      stepperSize = stepperSize,
+      text = text,
+      labelText = stepperStyle.labelText,
+      onMinusClick = onMinusClick,
+      onPlusClick = onPlusClick,
+      containerColor = containerColor.value,
+      textColor = textColor.value,
+      labelColor = labelColor.value,
+      plusColor = plusColor.value,
+      minusColor = minusColor.value,
+      errorText = errorText,
+      showError = showError,
+      modifier = modifier,
+    )
+  }
+}
+
+@Composable
+private fun DefaultStepper(
+  stepperSize: StepperDefaults.StepperSize,
+  text: String,
+  onMinusClick: () -> Unit,
+  onPlusClick: () -> Unit,
+  containerColor: Color,
+  textColor: Color,
+  plusColor: Color,
+  minusColor: Color,
+  showError: Boolean,
+  errorText: String?,
+  modifier: Modifier = Modifier,
+) {
+  Column(modifier) {
+    Surface(
+      shape = stepperSize.size.shape,
+      color = containerColor,
+      modifier = Modifier.defaultMinSize(minHeight = stepperSize.size.minHeight(Default)),
```

Comment:
Yeah I think I understand what you mean regarding the padding being hard to get right without the minHeight.

I faced the exact same thing when working on the text field.
So the only way to properly make this work without having to resort to minHeight, which imo works a bit weird when the text font size changes, is to have different paddings for when there is a label and where there is no label.

So it looks like for the Large component (first column) for no label it's: top = 16, bottom = 18
For the one with label: top = 10, bottom = 9.

Also a weird thing is that the two texts when there *is* a label are actually overlapping a bit. In that first it's an overlap of 3.dps for example. You can see how this was done in the text field here https://github.com/HedvigInsurance/android/blob/ad43926deb13033b11d98ecd22a19f8f85fcebf0/app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/internal/HedvigTextFieldImpl.kt#L346 It would be added to the column in line 253 if I am reading this correctly.

So after making these adjustments we'd be able to remove the minHeight modifier completely, and the component would look relatively the same regardless of text sizes. And you should continue to apply this padding to the entire content as you are already doing, meaning that if the text font is in fact small enough that the stepper symbols are bigger, then that's fine, those will then be the ones that determine the size of the container.

Reply:
And I just realize that you are doing the label/no label paddings separately exactly as I say already, so lmk if there's something else that was tricky.